src/openloco/windows/PlayerInfoPanel.cpp

```diff
@@ -0,0 +1,423 @@
+    static void on_update(window* w);
+    static CorporateRating performanceToRating(int16_t ax);
+
+    // 0x43AA4C
+    static void playerMouseDown(ui::window* self, widget_index widgetIndex)
+    {
+        auto args = FormatArguments();
+        _sortedCompanies.clear();
+
+        for (const auto& c : companymgr::companies())
+        {
+            if (!c.empty())
+            {
+                _sortedCompanies.push_back(&c);
+            }
+        }
+
+        sort(
+            _sortedCompanies.begin(),
+            _sortedCompanies.end(),
+            [](const company* a, const company* b) {
+                return a->performance_index > b->performance_index;
+            });
+
+        const string_id positionArray[15] = {
+            string_ids::position_1st,
+            string_ids::position_2nd,
+            string_ids::position_3rd,
+            string_ids::position_4th,
+            string_ids::position_5th,
+            string_ids::position_6th,
+            string_ids::position_7th,
+            string_ids::position_8th,
+            string_ids::position_9th,
+            string_ids::position_10th,
+            string_ids::position_11th,
+            string_ids::position_12th,
+            string_ids::position_13th,
+            string_ids::position_14th,
+            string_ids::position_15th,
+        };
+
+        int index = 0;
+        auto highlightIndex = -1;
+        for (auto company : _sortedCompanies)
+        {
+            auto competitorObj = objectmgr::get<competitor_object>(company->competitor_id);
+            auto rating = performanceToRating(company->performance_index);
+
+            args.push(positionArray[index]);
+            args.push(gfx::recolour(competitorObj->images[company->owner_emotion], company->colour.primary << 19));
+            args.push(company->name);
+            args.push<uint16_t>(0); // Needed after a user string id
+            args.push(company->performance_index);
+            args.push(_ratingNames[rating]);
+
+            dropdown::add(index, string_ids::dropdown_company_performance, args);
+
+            if (is_player_company(company->id()))
+            {
+                highlightIndex = index;
+            }
+
+            index++;
+        }
+
+        dropdown::add(index++, string_ids::dropdown_companies_list, image_ids::company_list_dropdown_icon);
+        dropdown::show_below(self, widgetIndex, index, 25);
+        if (highlightIndex != -1)
+        {
+            dropdown::set_highlighted_item(highlightIndex);
+        }
+        _113DC78 = _113DC78 | (1 << 1);
+    }
+
+    // 0x43AB87
+    static void playerDropdownClick(int16_t itemIndex)
+    {
+        if (itemIndex == -1)
+        {
+            itemIndex = dropdown::get_highlighted_item();
+        }
+
+        // If its index is bigger than the list then its the company list extra item
+        if (static_cast<uint16_t>(itemIndex) >= _sortedCompanies.size())
```

Comment:
Just fwiw, there is a notion of a 'neutral' company in the game. Probably the result of folded companies?